apis/container/v1beta1/gkecluster_types.go

```diff
@@ -0,0 +1,1685 @@
+
+	// SubnetworkRef references to a Subnetwork and retrieves its URI
+	// +optional
+	// +immutable
+	SubnetworkRef *SubnetworkURIReferencerForGKECluster `json:"subnetworkRef,omitempty" resource:"attributereferencer"`
+
+	// TierSettings: Cluster tier settings.
+	// +optional
+	// +immutable
+	TierSettings *TierSettings `json:"tierSettings,omitempty"`
+
+	// VerticalPodAutoscaling: Cluster-level Vertical Pod Autoscaling
+	// configuration.
+	// +optional
+	VerticalPodAutoscaling *VerticalPodAutoscaling `json:"verticalPodAutoscaling,omitempty"`
+
+	// WorkloadIdentityConfig: Configuration for the use of Kubernetes
+	// Service Accounts in GCP IAM
+	// policies.
+	// +optional
+	WorkloadIdentityConfig *WorkloadIdentityConfig `json:"workloadIdentityConfig,omitempty"`
+}
+
+// AddonsConfig is configuration for the addons that can be automatically
+// spun up in the
+// cluster, enabling additional functionality.
+type AddonsConfig struct {
+	// CloudRunConfig: Configuration for the Cloud Run addon. The
+	// `IstioConfig` addon must be
+	// enabled in order to enable Cloud Run addon. This option can only be
+	// enabled
+	// at cluster creation time.
+	CloudRunConfig *CloudRunConfig `json:"cloudRunConfig,omitempty"`
+
+	// HorizontalPodAutoscaling: Configuration for the horizontal pod
+	// autoscaling feature, which
+	// increases or decreases the number of replica pods a replication
+	// controller
+	// has based on the resource usage of the existing pods.
+	HorizontalPodAutoscaling *HorizontalPodAutoscaling `json:"horizontalPodAutoscaling,omitempty"`
+
+	// HTTpLoadBalancing: Configuration for the HTTP (L7) load balancing
+	// controller addon, which
+	// makes it easy to set up HTTP load balancers for services in a
+	// cluster.
+	HTTPLoadBalancing *HTTPLoadBalancing `json:"httpLoadBalancing,omitempty"`
+
+	// IstioConfig: Configuration for Istio, an open platform to connect,
+	// manage, and secure
+	// microservices.
+	IstioConfig *IstioConfig `json:"istioConfig,omitempty"`
+
+	// KubernetesDashboard: Configuration for the Kubernetes Dashboard.
+	// This addon is deprecated, and will be disabled in 1.15. It is
+	// recommended
+	// to use the Cloud Console to manage and monitor your Kubernetes
+	// clusters,
+	// workloads and applications. For more information,
+	// see:
+	// https://cloud.google.com/kubernetes-engine/docs/concepts/dashboar
+	// ds
+	KubernetesDashboard *KubernetesDashboard `json:"kubernetesDashboard,omitempty"`
+
+	// NetworkPolicyConfig: Configuration for NetworkPolicy. This only
+	// tracks whether the addon
+	// is enabled or not on the Master, it does not track whether network
+	// policy
+	// is enabled for the nodes.
+	NetworkPolicyConfig *NetworkPolicyConfig `json:"networkPolicyConfig,omitempty"`
+}
+
+// CloudRunConfig is configuration options for the Cloud Run feature.
+type CloudRunConfig struct {
+	// Disabled: Whether Cloud Run addon is enabled for this cluster.
+	Disabled *bool `json:"disabled"`
```

Comment:
This is one where we `ForceSendFields`. It could technically not be a pointer because its parent is a pointer and it is required if its parent is, but I updated all fields that required force sending to pointer types to be consistent.